lib/gel/gemfile_parser.rb

```diff
@@ -151,6 +158,12 @@ def add_gem(name, requirements, options)
       raise "Only git sources can specify a :branch" if options[:branch] && !options[:git]
       raise "Duplicate entry for gem #{name.inspect}" if @gems.assoc(name)
 
+      if options[:install_if]
```

Comment:
I wasn't sure if this processing belongs here or in `flatten` above. I placed it here for now, but kind of feel like `flatten` might be a better place for it. Maybe even add a `normalize_options` method that would be responsible for (guess what) normalizing option values, which is in turn called after merging the hashes from `flatten`. 

WDYT @matthewd ?

Reply:
I'm fine with this location for now, though I wonder how a nested construct is expected to behave, e.g.

```ruby
install_if -> { true } do
  install_if -> { false } do
    gem "foo", install_if: -> { true }
  end
end
```

I suspect the current "just use the value closest to the `gem` call" behaviour is probably right -- but _if_ we should be combining all the conditions, then `flatten` would be a better fit.

Reply:
Yes, I've realized this yesterday as I was trying to make the `install_if` test a bit more resilient. 

There is a problem regarding this for `platforms` and `group` as well, though, since Gel only uses the closest value to the `gem` call, as you said, but Bundler actually merges all the values.

So with this:
```ruby
group :test do
  gem "foo", group: :development
end
```
Gel would not consider `"foo"` gem for the `:test` environment, but Bundler would.

A potential solution to this should be applicable for `group`, `platforms` _and_ `install_if` so I am leaving that for another improvement if it is something the project wants to do.